agent/tool-scripts/pprof

```diff
@@ -113,76 +118,81 @@ ose_pprof() {
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt http://localhost:6060/debug/pprof/heap 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api.txt http://localhost:6061/debug/pprof/profile
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt http://localhost:6061/debug/pprof/heap 
-        else
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S).txt http://localhost:6060/debug/pprof/profile 
-            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt http://localhost:6060/debug/pprof/heap 
-        fi
-    done 
+    get_masters &
+    get_hosts
 } 
+
+get_masters() {
+    while true; do
+        i=1
+        for master in "${masters[@]}"; do
+            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_profile.txt http://localhost:6060/debug/pprof/profile &
+            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_controller_heap.txt http://localhost:6060/debug/pprof/heap &
+            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_api_profile.txt http://localhost:6061/debug/pprof/profile &
+            $tool_bin tool $tool -text -seconds=$interval -output $PPROF_TMPDIR/${master}/$(date +%Y-%m-%d:%H:%M:%S)_api_heap.txt http://localhost:6061/debug/pprof/heap &
+            if (( i++%${#masters[@]}==0 )); then
+                sleep ${interval}
+            fi
+        done
+    done
+}
+
+get_hosts() {
+    while true; do
+        j=1
+        for host in "${hosts[@]}"; do
+            oc get --raw /debug/pprof/profile?seconds=${interval} --server https://${host}:10250 > ${host}_profile && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_profile.txt ${host}_profile &
+            oc get --raw /debug/pprof/heap --server https://${host}:10250 > ${host}_heap && ${tool_bin} tool ${tool} -text -seconds=${interval} -output $PPROF_TMPDIR/${host}/$(date +%Y-%m-%d:%H:%M:%S)_heap.txt ${host}_heap &
+            if (( j++%${#hosts[@]}==0 )); then
+                sleep ${interval}
+            fi
+        done
+    done
+}
+
+## check if the inventory variable is set correctly
+if [[ ! -s "$inventory" ]] || [[ ! -f "$inventory" ]]; then
+	echo "Please check if the inventory file exists, is not empty and the inventory variable is not set to a directory"
+	exit 1
+fi
+
 case "$mode" in 
     install)
         check_install_rpm $tool_package 
-        ose_pprof
+        #ose_pprof
```

Comment:
I've commented out the scripted configuration changes to enable master pprof endpoints. If we are going to enable it, it should only do something if the config hasn't been changed before. Now the startup has to be carefully intitated as the master-api and master-controllers share a config so they cannot both be restarted after the config changes.

Reply:
adding @ekuric to review since he was going to make updates to pprof script to support OCP 3.10

Reply:
@wabouhamad Who told you this, and when?

Reply:
Just met with @ekuric this morning, did not know you were making updates already

Reply:
- This PR is ~9mo old from when we discovered that the pprof endpoints were going behind RBAC
- You asked me to [comment](https://bugzilla.redhat.com/show_bug.cgi?id=1572919#c15) in your BZ which I did last week

Reply:
Was just following up on BZ [comment](https://bugzilla.redhat.com/show_bug.cgi?id=1572919#c12) , did not know you were that far ahead.  No worries.  This is great, we can leverage what you have